11_notify/willshen8/pkg/puppy/lostpuppy_test.go

```diff
@@ -0,0 +1,71 @@
+package puppy
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func runTest(t *testing.T, payload []byte, expectedBody string, expectedHTTPCode int) {
+	req, err := http.NewRequest("POST", "/api/lostpuppy/", bytes.NewBuffer(payload))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// setup router
+	router := SetupRouter()
+
+	//setup routes
+	SetupLostPuppyRoutes(router)
+	// Setup recorder
+	rr := httptest.NewRecorder()
+
+	// Execute
+	router.ServeHTTP(rr, req)
+
+	// Get actual response
+	resp := rr.Result()
+	defer resp.Body.Close()
+	body, _ := ioutil.ReadAll(resp.Body)
```

Comment:
Don't ignore errors, especially in tests - it can lead you to false positives for the test. Use `require.NoError(t, err)` instead.

Reply:
Thanks, I tend not wanting to swallow the error. But when I was doing it, I didn't know how to test the error. But now I know.